source/vistas/core/graphics/raycaster.py

```diff
@@ -116,18 +116,32 @@ def __init__(self, origin=None, direction=None, near=None, far=None):
         self.far = far if far else numpy.inf
 
     def set_from_camera(self, coords: tuple, camera):
+        """ Update the Raycaster's ray to extend from the given Camera. """
+
         self.ray.origin = camera.get_position()
         self.ray.direction = camera.unproject(coords)
         self.ray.direction.normalize()
 
-    def intersect_object(self, obj) -> List[Renderable.Intersection]:
-        intersects = obj.raycast(self)
-        intersects.sort(key=lambda i: i.distance)
+    def intersect_object(self, coords, obj, camera) -> List[Intersection]:
+        """ Retrieve intersections, sorted in ascending distance, to a given Object3D. """
+
+        intersects = []
+        if issubclass(obj.__class__, Object3D):
```

Comment:
I'm not sure what I was thinking here... It probably can be just as you say.

Reply:
Ah, now I know why. `issubclass` accepts a `class`, not an instance of that class. So, we have to pass it the `__class__` of the instance to make sure it is derived from the `Object3D` class.

Reply:
That's my point. `isinstance` will do the same thing, but you can leave off the `.__class__` part.

```python
class Foo:
   ...

class Bar(Foo):
  ...

>>> b = Bar()
>>> isinstance(b, Foo)
True
```

Reply:
Alrighty, I'll use `isinstance`. I think I mainly was enforcing the fact that it should be a `subclass` of it, because not all `Object3D` objects are renderable.

Reply:
Ah, ok. If your intent is to allow subclasses and _only_ subclasses, then it's perfect as is.